### DISCUSSION TITLE
Does supabase '.in' function have a limit for how big the array you pass in can be if its not using a select?

### DISCUSSION THREAD

**Discussion by snaseri** [supabase/supabase#28257] [Questions]
Hello Everyone,
I know in oracle DB if you are not using a select statement inside an in clause and directly passing in a list e.g. ['Apple', 'Bananas', 'Peach'] there is a small limited numbers of items you can pass due to Oracles memory. I was wondering if this is the case in PostgresSQL and if so what is the limit?

Initially I wanted to do something like this:
![image](https://github.com/user-attachments/assets/075e0b3b-a58a-4f9a-a6b2-1ec97ed81566)

However, it seems this is not possible with the current API. So I am fetching my list of users with a separate call and will be passing in the array to the above '.in_' clause.

**Comment by GaryAustin1** [comment]
The internet has a limit on the length of a URL that will limit the length of .in for the REST API as all the values go in the URL.
It varies by I believe it can be as low as 2k.

You can send your values to an rpc function as it uses the body for the parameters and do your in with SQL code.  I don't believe there is a limit on the parameters in Postgres SQL except for performance.

**Reply by snaseri** [reply]
Thanks for the response.

Do you have an example of what you mean by using an rpc function?

My current understanding is i will have to rewrite the query that the supabase.select() will run adding in my in() clause using an rpc function.

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/reference/javascript/rpc is an rpc call to a postgres function.
I think this has links to examples... https://github.com/supabase/postgrest-js/issues/393#issuecomment-1404012263
